arbitrator/src/arbitrator_node.cpp

```diff
@@ -1,96 +1,153 @@
+#include "arbitrator_node.hpp"
 
-    double target_plan;
-    pnh.param("target_plan_duration", target_plan, 15.0);
-    arbitrator::TreePlanner tp{*cf, png, bss, ros::Duration(target_plan)};
 
-    double min_plan_duration;
-    pnh.param("min_plan_duration", min_plan_duration, 6.0);
-
-    double planning_frequency;
-    pnh.param("planning_frequency", planning_frequency, 1.0);
-
-    carma_wm::WMListener wml;
-    auto wm = wml.getWorldModel();
-
-    arbitrator::Arbitrator arbitrator{
-        &nh, 
-        &pnh, 
-        &sm, 
-        &ci, 
-        tp, 
-        ros::Duration(min_plan_duration),
-        ros::Rate(planning_frequency),
-        wm };
+namespace arbitrator
+{
+    struct Object
+    {   
+        int value = 0;
+        
+    };
+    ArbitratorNode::ArbitratorNode(const rclcpp::NodeOptions& options) : carma_ros2_utils::CarmaLifecycleNode(options) 
+    {
+        // Create initial config
+        config_ = Config();
+
+        config_.min_plan_duration = declare_parameter<double>("min_plan_duration", config_.min_plan_duration);
+        config_.target_plan_duration = declare_parameter<double>("target_plan_duration", config_.target_plan_duration);
+        config_.planning_frequency = declare_parameter<double>("planning_frequency", config_.planning_frequency);
+        config_.beam_width = declare_parameter<int>("beam_width", config_.beam_width);
+        config_.use_fixed_costs = declare_parameter<bool>("use_fixed_costs", config_.use_fixed_costs);
+        config_.plugin_priorities = plugin_priorities_map_from_json(declare_parameter<std::string>("plugin_priorities", ""));
+    }
+    
+    carma_ros2_utils::CallbackReturn ArbitratorNode::handle_on_configure(const rclcpp_lifecycle::State &)
+    {
+        // Handle dependency injection
+        arbitrator::CapabilitiesInterface ci(shared_from_this());
+        arbitrator::ArbitratorStateMachine sm;
+        
+        // Reset config
+        config_ = Config();
+        
+        get_parameter<double>("min_plan_duration", config_.min_plan_duration);
+        get_parameter<double>("target_plan_duration", config_.target_plan_duration);
+        get_parameter<double>("planning_frequency", config_.planning_frequency);
+        get_parameter<int>("beam_width", config_.beam_width);
+        get_parameter<bool>("use_fixed_costs", config_.use_fixed_costs);
+        std::string json_string;
+        get_parameter<std::string>("plugin_priorities", json_string);
+
+        config_.plugin_priorities = plugin_priorities_map_from_json(json_string);
+
+        RCLCPP_INFO_STREAM(get_logger(), "Arbitrator Loaded Params: " << config_);
+
+        std::shared_ptr<arbitrator::CostFunction> cf;
+        arbitrator::CostSystemCostFunction cscf = arbitrator::CostSystemCostFunction();
+
+        Object* o;
+        {
+            o = new Object();
+            o->value = 5;
+        }
+        RCLCPP_ERROR_STREAM(get_logger(), "AAAAAAAAAAA" << o->value);
```

Comment:
Why do we have this object struct? I'm guessing this log was just to test. Can we remove it?